### PR TITLE
Fixed load_state_dict() error

### PR DESCRIPTION
**Pull request by pranshu28**

### CODE REVIEW

**Review by jhuang265 — APPROVED** [review]
Just tested it and it seems to load everything properly. I'm not sure if this is just a temporary fix since maybe later on you might need to consider saving all the critical gradients when calling `state_dict()`, but for now I think you can leave it as is here.
